File: java/src/test/java/problems101/LinkedListCycleTest.java
package problems101;

import com.google.common.collect.Iterables;
import concepts.SinglyListNode;
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LinkedListCycleTest extends SampledTest {
    @Override
    public String getFileName() {
        return "linked_list_cycle.json";
    }

    @Test
    public void test() {
        for (LinkedListCycle solution : LinkedListCycle.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                SinglyListNode input = SinglyListNode.of(sample.input.head);
                if (sample.input.pos > -1) {
                    Iterables.getLast(input).next = Iterables.get(input, sample.input.pos);
                }
                assertWithMessage(sample.getMessage())
                    .that(solution.hasCycle(input))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Boolean> {
        public static class Input {
            public int[] head;
            public int pos;
        }
    }
}

File: javascript/__tests__/problems101_200/linked-list-cycle.test.js
import {strictEqual} from 'assert';
import SinglyListNode from '../../src/concepts/singly-list-node';
import SOLUTIONS from '../../src/problems101_200/linked-list-cycle';
import getSamples from '../../../testing/js/src/sample/samples';

describe(
    'LinkedListCycle',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach(sample => {
                      const head = SinglyListNode.of(...sample.input.head);
                      if (sample.input.pos > -1) {
                        head.iterateLast().next = head.iterateTo(sample.input.pos);
                      }
                      strictEqual(
                          solution.hasCycle(head),
                          sample.output,
                          sample.message,
                      );
                    }),
                ),
        ),
);
